valmlord/gulpify
# feat: add SVG sprite compilation and usage in templates

## Description
This pull request enhances the Gulp boilerplate workflow by adding tasks specifically designed for compiling SVG sprites and using them in templates. It includes the integration of `gulp-svg-sprite` and the creation of a Pug mixin for including SVG icons.

## Changes
- Added `gulp-svg-sprite` package to `package.json`.
- Created a new Gulp task to compile SVG sprites using `gulp-svg-sprite`.
- Configured SVGO for optimized SVG processing while preserving important attributes.
- Created a Pug mixin for including SVG icons in templates.
- Updated `index.pug` to use the new SVG icon mixin.
- Ensured complex SVGs like logos are displayed correctly by adjusting SVGO settings.

## How to Test
1. Place SVG files in the appropriate directory `app/assets/icons`.
2. Execute the relevant Gulp task `gulp svgSprite` to compile the SVG sprites.
3. Verify that the compiled sprite is correctly generated in the `build/assets/icons` directory.
4. Include the generated sprite in your HTML or Pug templates and use the Pug mixin to include specific icons.

## Additional Notes
- This update aims to improve the efficiency of handling SVG icons by compiling them into a single sprite and using them in templates with a Pug mixin.
- The SVGO configuration has been adjusted to ensure that complex SVGs, such as logos, are displayed correctly without losing important attributes.
